sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,555 @@
+            return np.concatenate([X] + [pred.reshape(-1, 1) if pred.ndim == 1
+                                         else pred for pred in y_pred], axis=1)
+        else:
+            return np.concatenate([pred.reshape(-1, 1) if pred.ndim == 1
+                                   else pred for pred in y_pred], axis=1)
+
+    def set_params(self, **params):
+        """ Setting the parameters for the stacking estimator.
+
+        Valid parameter keys can be listed with get_params().
+
+        Parameters
+        ----------
+        params : keyword arguments
+            Specific parameters using e.g. set_params(parameter_name=new_value)
+            In addition, to setting the parameters of the ``VotingClassifier``,
+            the individual classifiers of the ``VotingClassifier`` can also be
+            set or replaced by setting them to None.
+
+        Examples
+        --------
+        # In this example, the RandomForestClassifier is removed
+        clf1 = LogisticRegression()
+        clf2 = RandomForestClassifier()
+        eclf = StackingClassifier(estimators=[('lr', clf1), ('rf', clf2)]
+        eclf.set_params(rf=None)
+
+        """
+        super()._set_params('estimators', **params)
+        return self
+
+    def get_params(self, deep=True):
+        """ Get the parameters of the stacking estimator.
+
+        Parameters
+        ----------
+        deep: bool
+            Setting it to True gets the various classifiers and the parameters
+            of the classifiers as well.
+        """
+        return super()._get_params('estimators', deep=deep)
+
+    def fit(self, X, y, sample_weight=None):
+        """ Fit the estimators.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape (n_samples,)
+            Target values.
+
+        sample_weight : array-like, shape (n_samples,) or None
+            Sample weights. If None, then samples are equally weighted.
+            Note that this is supported only if all underlying estimators
+            support sample weights.
+
+        Returns
+        -------
+        self : object
+        """
+
+        self._validate_meta_estimator()
+
+        if self.estimators is None or len(self.estimators) == 0:
+            raise AttributeError(
+                "Invalid 'estimators' attribute, 'estimators' should be a list"
+                " of (string, estimator) tuples."
+            )
+
+        if not isinstance(self.passthrough, bool):
+            raise AttributeError(
+                "Invalid 'passthrough' attribute, 'passthrough' should be a "
+                "boolean. Got {!r} instead.".format(self.passthrough)
+            )
+
+        names, estimators_ = zip(*self.estimators)
+        self._validate_names(names)
+
+        n_isnone = np.sum(
```

Comment:
I'd suggest not using numpy, and not even create the list: and iterator is enough